backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/codec/impl/ElasticsearchMonthDayFieldCodec.java

```diff
@@ -41,4 +42,10 @@ protected MonthDay nullUnsafeParse(String stringValue) {
 		LocalDate date = LocalDate.parse( stringValue, formatter );
 		return MonthDay.from( date );
 	}
+
+	@Override
+	protected Long nullUnsafeScalar(MonthDay value) {
+		LocalDate date = value.atYear( LEAP_YEAR );
+		return date.atStartOfDay().atZone( ZoneOffset.UTC ).toInstant().toEpochMilli();
```

Comment:
You can just call `value.toInstant( ZoneOffset.UTC )` instead of `atZone(...).toInstant()`.